src/profile-logic/profile-data.js

```diff
@@ -417,86 +417,97 @@ export function collapsePlatformStackFrames(thread: Thread): Thread {
   });
 }
 
+export function filterThreadToSearchStrings(
+  thread: Thread,
+  searchStrings: string[]
+): Thread {
+  return timeCode('filterThreadToSearchStrings', () => {
+    if (!searchStrings.length) {
+      return thread;
+    }
+
+    return searchStrings.reduce(filterThreadToSearchString, thread);
+  });
+}
+
 export function filterThreadToSearchString(
   thread: Thread,
   searchString: string
 ): Thread {
-  return timeCode('filterThreadToSearchString', () => {
-    if (searchString === '') {
-      return thread;
+  if (!searchString) {
```

Comment:
NOTE: Actually nothing changed in this function, only the indentation and the removal of the timeCode call, moved to the function filterThreadToSearchStrings above.